source/Calamari.Common/Plumbing/Extensions/EnumerableExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Generic;
+
+namespace Calamari.Common.Plumbing.Extensions
+{
+    public static class EnumerableExtensions
+    {
+        internal static IEnumerable<TSource> DistinctBy<TSource, TKey>(this IEnumerable<TSource> source,
```

Comment:
This function exists in .net 6 but not in .Net Framework 4.0 (or 4.5.2). This implementation was at the end of `Glob.cs` so I've moved it here.